code/modules/mob/living/living_defense.dm

```diff
@@ -396,7 +396,10 @@
 /mob/living/proc/flash_act(intensity = 1, override_blindness_check = 0, affect_silicon = 0, visual = 0, type = /atom/movable/screen/fullscreen/flash)
 	if(get_eye_protection() < intensity && (override_blindness_check || !(HAS_TRAIT(src, TRAIT_BLIND))))
 		overlay_fullscreen("flash", type)
-		addtimer(CALLBACK(src, .proc/clear_fullscreen, "flash", 25), 25)
+		if(issilicon(src))
+			addtimer(CALLBACK(src, .proc/clear_fullscreen, "flash", 25), 70) //Flashes now blind silicons for the duration they used to stun
```

Comment:
Override /mob/living/silicon/flash_act instead of checking for silicons in living mobs flash_act

Reply:
👍